Sources/TensorFlow/Operators/Math.swift

```diff
@@ -1505,25 +1505,59 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
 
 /// Performs matrix multiplication with another tensor and produces the result.
 @inlinable
-@differentiable(vjp: _vjpMatmul(_:_:) where Scalar: TensorFlowFloatingPoint)
+@differentiable(vjp: _vjpMatmul(_:_:transposeA:transposeB:) where Scalar: TensorFlowFloatingPoint)
 public func matmul<Scalar: Numeric>(
     _ lhs: Tensor<Scalar>,
-    _ rhs: Tensor<Scalar>
+    _ rhs: Tensor<Scalar>,
+    transposeA: Bool = false,
+    transposeB: Bool = false
 ) -> Tensor<Scalar> {
-    // Default arguments specified explicitly to avoid "external declarations of SILFunctions with
-    // shared visibility is not allowed" SILVerifier error in
-    // "tests/AutoDiff/tensor_autodiff_runtime.swift".
-    return Raw.matMul(lhs, rhs, transposeA: false, transposeB: false)
+    if lhs.rank > 2 && rhs.rank > 2 {
```

Comment:
How about

```swift
switch (lhs.rank, rhs.rank) {
case (3..., 3...):
  ...
case (2, 3...):
  ...
case (3..., 2):
  ...
default:
  ...
}
```

Reply:
I didn't know you could pattern match on integers with ranges. That's cool! :)